app/assets/javascripts/app/helpers/text_formatter.js

```diff
@@ -1,146 +1,76 @@
+
+      url = // rebuild the url
+        (!addr.scheme ? '' : addr.scheme +
+        ( (addr.scheme.toLowerCase()=="mailto") ? ':' : '://')) +
+        (!addr.user ? '' : addr.user +
+        (!addr.pass ? '' : ':'+addr.pass) + '@') +
+        punycode.toASCII(addr.host) +
+        (!addr.port ? '' : ':' + addr.port) +
+        (!addr.path ? '' : encodeURI(addr.path) ) +
+        (!addr.query ? '' : '?' + encodeURI(addr.query) ) +
+        (!addr.fragment ? '' : '#' + encodeURI(addr.fragment) );
+      return url;
+    };
+
+    var md = window.markdownit({
+      breaks:      true,
+      html:        false,
+      linkify:     true,
+      typographer: true
     });
+    var iterator = window.markdownitForInline;
 
-    // make nice little utf-8 symbols
-    converter.hooks.chain("preConversion", function(text) {
-      var input_strings = [
-        "<->", "->", "<-",
-        "(c)", "(r)", "(tm)",
-        "<3"
-      ];
-      var output_symbols = [
-        "↔", "→", "←",
-        "©", "®", "™",
-        "♥"
-      ];
-      // quote function from: http://stackoverflow.com/a/494122
-      var quote = function(str) {
-        return str.replace(/([.?*+^$[\]\\(){}|-])/g, "\\$1");
-      };
-
-      _.each(input_strings, function(str, idx) {
-        var r = new RegExp(quote(str), "gi");
-        text = text.replace(r, output_symbols[idx]);
-      });
-      return text;
+    md.use(iterator, 'utf8_symbols', 'text', function (tokens, idx) {
+      tokens[idx].content = tokens[idx].content.replace(/<->/g, "↔")
+                                               .replace(/<-/g,  "←")
+                                               .replace(/->/g,  "→")
+                                               .replace(/<3/g,  "♥");
     });
 
-    converter.hooks.chain("postConversion", function (text) {
-      return text.replace(/(\"(?:(?:http|https):\/\/)?[a-zA-Z0-9\-\.]+\.[a-zA-Z]{2,3}(?:\/\S*)?\")(\>)/g, '$1 target="_blank">')
+    md.use(iterator, 'link_new_window_and_punycode', 'link_open', function (tokens, idx) {
+      tokens[idx].href = punycodeURL(tokens[idx].href);
+      tokens[idx].target = "_blank";
     });
 
-    return converter.makeHtml(text)
-  };
-
-  textFormatter.hashtagify = function hashtagify(text){
-    var utf8WordCharcters =/(<a[^>]*>.*?<\/a>)|(\s|^|>)#([\u0080-\uFFFF|\w|-]+|&lt;3)/g;
+    md.use(iterator, 'image_punycode', 'image', function (tokens, idx) {
+      tokens[idx].src = punycodeURL(tokens[idx].src);
+    });
 
-    return text.replace(utf8WordCharcters, function(result, linkTag, preceeder, tagText) {
-      if(linkTag)
-        return linkTag;
-      else
-        return preceeder + "<a href='/tags/" + tagText.toLowerCase() +
-                           "' class='tag'>#" + tagText + "</a>";
+    var hashtagPlugin = window.markdownitHashtag;
+    md.use(hashtagPlugin, {
+      // TODO is this [:alnum:] in ruby?
+      hashtagRegExp: '[\\x0080-\\xFFFF\\w\\-]+|<3',
```

Comment:
They're documented at the  [Regexp class](http://ruby-doc.org/core-2.2.0/Regexp.html) in the Ruby documentation:

> POSIX bracket expressions are also similar to character classes. They provide a portable alternative to the above, with the added benefit that they encompass non-ASCII characters. For instance, /\d/ matches only the ASCII decimal digits (0-9); whereas /[[:digit:]]/ matches any character in the Unicode Nd category.

So on [into the standard](http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap09.html#tag_09_03_05):

> The set of single-character collating elements whose characters belong to the character class, as defined in the LC_CTYPE category in the current locale.

Oh well not too helpful. Source code diving time then! The engine used by Ruby for regular expressions is called oniguruma. Some digging into its code brings up: https://github.com/ruby/ruby/blob/trunk/enc/unicode/name2ctype.src#L3695

Uhm okay, someone had fun figuring that out there. 566, looks like the number of entries. Well, actually half of it, so we're dealing with pairs here. Talking about codepoints, must be ranges.

Onto some Ruby magic! [Let's crash everyones browser](http://paste.mrzyx.de/pzxah2po4)!

Now that was fun, wasn't it?